core/src/main/scala/org/apache/spark/storage/DiskBlockObjectWriter.scala

```diff
@@ -262,6 +262,27 @@ private[spark] class DiskBlockObjectWriter(
     file
   }
 
+  /**
+   * Reverts write metrics that haven't been committed yet and delete the file held
+   * by current `DiskBlockObjectWriter`. Callers should invoke this function when there
+   * are runtime exceptions in file writing process and the file is no longer needed.
+   */
+  def deleteHeldFile(): Unit = {
```

Comment:
@Ngone51 I compromised and encapsulated the process of deleting the file held by `DiskBlockObjectWriter` into this method, haha

Reply:
rename to `closeAndDelete` ?

Reply:
ok